UI/Web/src/app/manga-reader/_components/single-renderer/single-renderer.component.ts

```diff
@@ -67,6 +74,28 @@ export class SingleRendererComponent implements OnInit, ImageRenderer {
       takeUntilDestroyed(this.destroyRef)
     );
 
+    //handle manual width
+    this.mangaReaderComponent.generalSettingsForm.get("widthSlider")?.valueChanges.pipe(takeUntilDestroyed(this.destroyRef)).subscribe(val => {
+      val = (val < 0) ? 'none' : val + "%";
+      this.widthOverride$ = val;
+
+      var img = this.document.getElementById('image-1');
+      if(!img) return;
+
+      var scaling = this.mangaReaderComponent.generalSettingsForm.get("fittingOption")?.value;
+      if (FITTING_OPTION.HEIGHT == scaling) return;
+
+      if(this.widthOverride$ == 'none') img.style.width = "";
```

Comment:
We don't need to set things on the image like this. This should map to a width value (percentage or none) and then that can be set directly on the <img> tag with style="width: {{widthOverride$ | async}}". This will then be "width: none" or "width: 20%"

Reply:
can you explain this further? I am not experienced in this area and when I try this like you say I get "cannot assign to 'style' because it is an import.

Reply:
The idea is that in Angular we don't set things on DOM nodes directly, so instead of setting width like this through logic, instead we want to store the width value in an observable (widthOverride$) then in the template, directly use that value like I showed above. 

You can see something similar for imageFitClass$.

Reply:
I see, as someone normally only doing procedural programming in C++ this whole webdev angular stuff is very confusing to me. Hopefully I get the code to your liking the next time
The logic itself should work fine, even deployed it on my own server.